core/common/token-core/src/test/java/org/eclipse/edc/token/TokenDecoratorRegistryImplTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.token;
+
+import org.eclipse.edc.token.spi.TokenDecorator;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+class TokenDecoratorRegistryImplTest {
+
+    private final TokenDecoratorRegistryImpl tokenDecoratorRegistry = new TokenDecoratorRegistryImpl();
+
+    @Test
+    void register_whenContextNotExists() {
+        tokenDecoratorRegistry.register("test-context", (claims, headers) -> {
+        });
+
+        assertThat(tokenDecoratorRegistry.getDecoratorsFor("test-context")).hasSize(1);
+    }
+
+    @Test
+    void register_whenContextExists() {
```

Comment:
that's not exactly true, because here, two different objects (different object references) is passed, whereas in the next test, the same object is passed.

Reply:
and what's the practical difference? the registry implementation does not care if the same decorator is registered twice